### PR TITLE
Tabular: Added random_seed parameter

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*
#425 

*Description of changes:*

Added random_seed parameter to TabularPrediction. Functionality is identical to previous if random_seed is kept as default value.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-429-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-429/1/index.html

File: autogluon/utils/tabular/ml/trainer/auto_trainer.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def train(self, X_train, y_train, X_test=None, y_test=None, hyperparameter_tune=
             y_test = None
         else:
             if (y_test is None) or (X_test is None):
-                X_train, X_test, y_train, y_test = generate_train_test_split(X_train, y_train, problem_type=self.problem_type, test_size=holdout_frac)
+                X_train, X_test, y_train, y_test = generate_train_test_split(X_train, y_train, problem_type=self.problem_type, test_size=holdout_frac, random_state=self.random_seed)
```
Comment (jwmueller) [inline autogluon/utils/tabular/ml/trainer/auto_trainer.py:38]: Default `random_state` in `generate_train_test_split` is currently 42? Its default value should probably match the `random_seed` default

**Comment by jwmueller** [comment]
This PR is not intended to fully address the reproducibility issue, right?  If so I'm happy to approve it as just a start.  But if this PR is intended to fully address the linked issue, then the `random_seed` arg of fit() needs to be passed to each of the models, no?

File: autogluon/utils/tabular/ml/trainer/auto_trainer.py (hunk 1 above)
Comment (Innixma) [reply autogluon/utils/tabular/ml/trainer/auto_trainer.py:38]: The default value in the function itself is always overridden, so there is no impact, but I will change it to 0... @sgdread So long and thanks for all the fish.

**Comment by Innixma** [comment]
> This PR is not intended to fully address the reproducibility issue, right? If so I'm happy to approve it as just a start. But if this PR is intended to fully address the linked issue, then the `random_seed` arg of fit() needs to be passed to each of the models, no?

Initially I wasn't sure what the exact request was in the issue. This is only intended to enable reproducible data splits (and the ability to randomize them each run, prior it was always the same splits). This does not fully address the linked issue, which is actually more complex than propagating the random_seed value to the models.

**Comment by szha** [comment]
Job PR-429-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-429/2/index.html

**Comment by jwmueller** [comment]
> This is only intended to enable reproducible data splits (and the ability to randomize them each run, prior it was always the same splits). This does not fully address the linked issue, which is actually more complex than propagating the random_seed value to the models.

Right, I'd just make sure to add a cautionary sentence to the documentation here:  
https://github.com/awslabs/autogluon/pull/429/files#diff-fedbb5dec476a83cdce171966541d6a7R257

Otherwise users who don't carefully read it will think this can achieve fully reproducible fit().  For example we could say:

"Caution: This seed only enables reproducible data splits (and the ability to randomize splits in  each run by changing seed values).  This seed is NOT used in the training of individual models, for that you need explicitly set the corresponding seed hyperparameter (usually called 'seed_value') of each individual model."

**Comment by Innixma** [comment]
Added additional warning in documentation as requested.

**Comment by szha** [comment]
Job PR-429-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-429/3/index.html

**Comment by amirsani** [comment]
Would the following be the proper way to use this if I want to set the seeds for each fit,

``
i = 0

hyperparameters = {
               #'NN': {},
               'GBM': {"seed":i},
               'CAT': {"random_seed":i},
               'RF': {"random_state":i},
               'XT': {"random_state":i},
               'KNN': {"random_state":i},
               #'custom': ['GBM']
} 

task.fit(..., hyperparameters=hyperparameters, random_seed=i)
``

Thank you

**Comment by Innixma** [comment]
@amirsani This looks correct to me
